crates/detect-targets/src/detect/windows.rs

```diff
@@ -1,6 +1,40 @@
-pub(super) fn detect_alternative_targets(target: &str) -> Option<String> {
-    let (prefix, abi) = target.rsplit_once('-')?;
+use windows::Win32::System::{
+    SystemInformation::{IMAGE_FILE_MACHINE, IMAGE_FILE_MACHINE_AMD64},
+    Threading::{GetMachineTypeAttributes, UserEnabled, Wow64Container, MACHINE_ATTRIBUTES},
+};
+
+trait BitMaskExt {
+    fn is_set(self, oth: Self) -> bool;
+}
+
+impl BitMaskExt for MACHINE_ATTRIBUTES {
+    fn is_set(self, oth: Self) -> bool {
+        // MACHINE_ATTRIBUTES does not implement PartialEq<u32>,
+        // so I have to obtain its inner u32 and compare it manually.
+        (self & oth).0 != 0
+    }
+}
+
+fn is_arch_supported(arch: IMAGE_FILE_MACHINE) -> bool {
+    // SAFETY: GetMachineTypeAttributes takes any argument with type
+    // IMAGE_FILE_MACHINE.
+    unsafe { GetMachineTypeAttributes(arch.0) }
+        .ok()
+        .map(|attrs| attrs.is_set(Wow64Container) || attrs.is_set(UserEnabled))
+        .unwrap_or(false)
+}
+
+pub(super) fn detect_alternative_targets(target: &str) -> impl Iterator<Item = String> {
+    let (prefix, abi) = target
+        .rsplit_once('-')
+        .expect("unwrap: target always has a -");
 
     // detect abi in ["gnu", "gnullvm", ...]
-    (abi != "msvc").then(|| format!("{prefix}-msvc"))
+    let msvc_fallback_target = (abi != "msvc").then(|| format!("{prefix}-msvc"));
+
+    let x64_fallback_target = (!prefix.starts_with("x86_64")
+        && is_arch_supported(IMAGE_FILE_MACHINE_AMD64))
+    .then(|| "x86_64-pc-windows-msvc".to_string());
```

Comment:
We could also do the other ones while we're at it:
- x86 (on both x64 and arm)
- maybe the 32-bit arms

Reply:
I've added fallback for both x86 and arm32